python/bind_data_access.h

```diff
@@ -70,17 +70,24 @@ class DataAccessHelper {
         return py::dtype::of<T>();
       }
     };
+    const auto get_base = [&view]() {
+      if constexpr (std::is_same_v<std::decay_t<View>, scipp::Variable>) {
+        return py::cast(view.data_handle());
+      } else {
+        return py::cast(view.data().data_handle());
+      }
+    };
     const auto &dims = view.dims();
     if (view.is_readonly()) {
-      auto array =
-          py::array{get_dtype(), dims.shape(), get_strides(),
-                    Getter::template get<T>(std::as_const(view)).data(), obj};
+      auto array = py::array{
+          get_dtype(), dims.shape(), get_strides(),
+          Getter::template get<T>(std::as_const(view)).data(), get_base()};
```

Comment:
Hmm, interesting, does this have the additional (probably minor/irrelevant) advantage that gettings `values` of a `DataArray` now does not unnecessarily keep meta data alive?

Reply:
How did it work before, did the Variable keep the whole Dict alive? If not, then we are only saving very little here.

Reply:
Yes, the `values` kept the whole `DataArray` (and potentially `Dataset`) including their dicts alive.